server/src/main/java/org/apache/iotdb/db/service/DataNode.java

```diff
@@ -134,6 +147,8 @@ protected void doAddNode() {
       prepareDataNode();
       // register current DataNode to ConfigNode
       registerInConfigNode();
+      // get resources for trigger,udf...
+      prepareResources();
```

Comment:
Do we have any tests on real cluster? I'm not sure if there exists problems if put `prepareResources()` ahead of `active()` and `setUpRPCService()`

Reply:
I have tested it manually and I think it is ok? I moved this method into active() now.